### DISCUSSION TITLE
自动记账4.0公测开始啦！！！

### DISCUSSION THREAD

**Discussion by AnkioTomas** [AutoAccountingOrg/AutoAccounting#1] [公告]
# 版本

目前最新的公测版本为：beta8

# 警告

目前未发布正式版，测试版可能导致你的手机出现高耗电、卡顿、高内存占用等问题，请慎重尝鲜。

# 说明
- 本次测试不会影响原有的自动记账App；
- 本次测试仅针对`Xposed`用户（Lsposed、虚拟框架）；
- [`钱迹补丁`](https://github.com/AutoAccountingOrg/QianjiPatch)插件还未开发完成，因此暂时无法正常记录账单（记录到记账软件来），可以记录到本地来；
- 由于缺少规则，部分账单无法识别，请使用`上传数据`功能上传到github等待适配；
> 部分小伙伴把无关数据上传了，建议上传数据之前点开看看是不是交易信息

# 规则
- 自动记账4.0的识别规则由官方统一维护，缓存时间1小时，适配后等更新即可；
- [规则仓库](https://github.com/AutoAccountingOrg/AutoRule)在此，有需要可直接提PR；

# Hook应用
 - `支付宝`
 - `微信`

# 适配
- 第三方App的适配还要等一段时间，文档还没准备好；
- 如果着急适配可以参考[demo](https://github.com/AutoAccountingOrg/AutoAccounting/blob/master/demo/src/main/java/net/ankio/demo/MainActivity.kt)，需要引入sdk依赖：`com.github.AutoAccountingOrg.AutoAccounting:sdk:-SNAPSHOT`，如果需要自动记账自己定义的对象，还需要引入``com.github.AutoAccountingOrg.AutoAccounting:common:-SNAPSHOT``，当然也可以自己写通信过程。

# 下载

[阿里云盘/持续构建板](https://cloud.ankio.net/%E9%98%BF%E9%87%8C%E4%BA%91%E7%9B%98/%E8%87%AA%E5%8A%A8%E8%AE%B0%E8%B4%A6/%E7%89%88%E6%9C%AC%E6%9B%B4%E6%96%B0/%E6%8C%81%E7%BB%AD%E6%9E%84%E5%BB%BA%E7%89%88)

# 反馈

[Bug/功能请求](https://github.com/AutoAccountingOrg/AutoAccounting/issues)

[讨论/闲聊](https://github.com/orgs/AutoAccountingOrg/discussions)

**Comment by MIONKING** [comment]
🥳大佬加油，冲冲冲

**Comment by shilei-development** [comment]
老大，启动出问题了
![Screenshot_2024-03-22-14-09-11-047_net ankio auto xposed](https://github.com/AutoAccountingOrg/AutoAccounting/assets/163731738/d32ec50b-e77a-4021-9df9-ac9ae80542a6)

**Reply by AnkioTomas** [reply]
等修，github的锅

**Reply by shilei-development** [reply]
👌🏻

**Comment by wangjinbaor** [comment]
![Screenshot_2024-04-09-19-58-58-44_a87fd7db6caa850b517aa6fa9d2fcd0e.jpg](https://github.com/AutoAccountingOrg/AutoAccounting/assets/58154171/80e04f00-5f02-404a-82a9-25a2288e5d43)

老大， b3 的链接好像有问题

**Reply by AnkioTomas** [reply]
恢复了

**Comment by Jonekadsen** [comment]
每天上来看一看，看老大一点点写（虽说说的下半年也不定是哪个下半年😜

**Comment by JackJia000** [comment]
微信云端规则，转账0.01，会被识别成1元，0.02识别成2元，还没发修改规则，跟识别![S40502-07101694_net.ankio.auto.xposed.png](https://github.com/AutoAccountingOrg/AutoAccounting/assets/94772782/fcffbad6-5abc-4ed1-91d0-edb5fda8bdd2)

结果

**Reply by AnkioTomas** [reply]
已更新v0.2.1的规则修复该问题

**Comment by JackJia000** [comment]
![S40503-15021068_net.ankio.auto.xposed(1).png](https://github.com/AutoAccountingOrg/AutoAccounting/assets/94772782/390bcb22-5ddd-4647-b37c-f0420b97aa24)


![S40503-15014485_net.ankio.auto.xposed(1).png](https://github.com/AutoAccountingOrg/AutoAccounting/assets/94772782/f6c43f57-5418-4406-aa12-6cc416b67d25)

**Reply by JackJia000** [reply]
![S40503-15021735_net.ankio.auto.xposed(1).png](https://github.com/AutoAccountingOrg/AutoAccounting/assets/94772782/03ed5b50-fa25-4715-9416-153bb397fb5e)

**Reply by shenyoujunjue** [reply]
哥为什么你可以用钱迹和自动记账，我的不识别？

**Reply by JackJia000** [reply]
1.需要三个软件.钱迹/钱迹补丁/自动记账
2.下载安装完，lsp设置作用域，
3.打开钱迹app，左上侧三道杠，找到钱迹补丁，修补
4.打开自动记账app，启动，同步钱迹的信息，账户映射。等到自己想把账单同步到钱迹时候，手动同步等等，大概这样吧
个人感觉：4.0增加了去重功能，其他功能不如3系列

**Reply by JackJia000** [reply]
> 1.需要三个软件.钱迹/钱迹补丁/自动记账
> 2.下载安装完，lsp设置作用域，
> 3.打开钱迹app，左上侧三道杠，找到钱迹补丁，修补
> 4.打开自动记账app，启动，同步钱迹的信息，账户映射。等到自己想把账单同步到钱迹时候，手动同步等等，大概这样吧
> 个人感觉：4.0增加了去重功能，其他功能不如3系列
> 

如果能减少手动操作的动作就好很多了，比①如启动一次，下次就不用手动启动了。(目前是动不动就不启动了，时不时都要去看有没有在运行，这也是劝退我使用3系列主要原因)
②匹配到规则自动添加到账单页面。
③可以设置多长时间自动同步账单到钱迹。
这样应该就相当不错了
个人非开发者不知道这些实现起来难度大不大，希望越来越好用

**Comment by MisakaSakura** [comment]
我看钱迹补丁那边已经自动构建apk了，为啥自动记账还是不能识别到钱迹呢？![Screenshot_2024-05-10-18-44-29-35_696fc9dfea9e2f51729e0646b2031c10.jpg](https://github.com/AutoAccountingOrg/AutoAccounting/assets/30547550/c9789fb2-e0d3-49d6-81b5-0ff871f1d525)

**Comment by AnkioTomas** [comment]
你需要在钱迹里面打开自动记账功能【左侧 - 钱迹补丁】
—————————
Ankio 敬上
***@***.***
若无闲事挂心头，便是人间好时节。

**Reply by pwupink** [reply]
找不到该选项，请提供截图或视频教程，感谢

**Reply by justcopy** [reply]
你要先安装钱迹补丁，然后lsp勾选

**Comment by 3170807233** [comment]
免密支付没有被记录![Screenshot_2024-05-20-20-42-59-701_com.eg.android.AlipayGphone-edit.jpg](https://github.com/AutoAccountingOrg/AutoAccounting/assets/59640129/579bb35f-7886-46a7-a781-0b58092c018a)

![Screenshot_2024-05-20-20-44-20-562_net.ankio.auto.xposed.jpg](https://github.com/AutoAccountingOrg/AutoAccounting/assets/59640129/9ba4aeaf-c373-41cc-95c1-9a88be508e77)

**Comment by mcthesw** [comment]
支持大佬！

**Comment by DreamSWL** [comment]
![Screenshot_2024-06-20-21-25-53-420_com mutangtech qianji](https://github.com/AutoAccountingOrg/AutoAccounting/assets/33590189/2b5f42ff-c436-4fca-acbe-02fc14e0f478)

![Screenshot_2024-06-20-21-25-39-305_net ankio auto xposed](https://github.com/AutoAccountingOrg/AutoAccounting/assets/33590189/5fcf992a-f4e3-401f-8acf-994a0a2cfb9d)
不显示资产数据。LSP里面重新优化也点了

**Comment by LX1639193100** [comment]
![Screenshot_2024-06-29-00-50-15-01_696fc9dfea9e2f51729e0646b2031c10](https://github.com/AutoAccountingOrg/AutoAccounting/assets/103544608/d60a996f-894f-42b4-b69c-d7a485d16259)
服务 root 启动失败

**Reply by AnkioTomas** [reply]
这是BUG，等修

**Comment by zyyyy68** [comment]
![Screenshot_20240720_213300](https://github.com/user-attachments/assets/9a6de971-deca-4f5d-ab2c-2132d3b16483)
适配那个版本啊

**Comment by Hades-ming** [comment]
大佬 alist网盘失效了  
![image](https://github.com/user-attachments/assets/4d95538e-7443-45de-a0b5-fb70bfd9b1c8)
